core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -1002,4 +1014,38 @@ public void createCronForUserHomeCreation(StackDto stackDto, Set<String> candida
             throw new CloudbreakException(message, e);
         }
     }
+
+    private void addRangerRazServiceIfAvailable(StackDto stackDto, String cmVersion, Map<String, List<String>> serviceLocations, GatewayConfig gatewayConfig) {
+        if (stackDto.getCluster() != null && stackDto.getStack() != null
+                && CMRepositoryVersionUtil.isRazConfigurationSupported(
+                cmVersion, CloudPlatform.valueOf(stackDto.getCloudPlatform()), stackDto.getStack().getType())
+        ) {
+            boolean razEnabled = stackDto.getCluster().isRangerRazEnabled();
+
+            if (StringUtils.isNotEmpty(stackDto.getEnvironmentCrn()) && StackType.WORKLOAD.equals(stackDto.getStack().getType())) {
+                List<SdxClusterResponse> datalakes = sdxClientService.getByEnvironmentCrn(stackDto.getEnvironmentCrn());
+                if (!datalakes.isEmpty()) {
+                    razEnabled = datalakes.get(0).getRangerRazEnabled();
+                }
+            }
+
+            if (razEnabled) {
+                ExposedService rangerRazService = exposedServiceCollector.getRangerRazService();
+
+                // Find master nodes since Ranger Raz is only going to be installed on master nodes
+                List<String> rangerRazLocations = stackDto.getAliveInstancesInInstanceGroup("master").stream()
```

Comment:
I think you should probably select the GATEWAY node right ?

Reply:
Done